packages/tldraw/src/lib/shapes/line/LineShapeUtil.tsx

```diff
@@ -185,6 +186,23 @@ export class LineShapeUtil extends ShapeUtil<TLLineShape> {
 			},
 		}
 	}
+	override getInterpolatedProps(
+		startShape: TLLineShape,
+		endShape: TLLineShape,
+		progress: number
+	): TLLineShape['props'] {
+		return {
+			...endShape.props,
+			points: {
+				...mapObjectMapValues(startShape.props.points, (_, { id, index, x, y }) => ({
```

Comment:
I believe this will give spooky NaN values if there are less points in the end shape than in the start shape.

If you wanted to get the multi-point lines working, I think the way I would approach this would be to first say that if the spline is cubic and the start/end don't have the same number of point, we skip lerping entirely. cubic splines are spooky, and we don't have enough information in our props to interpolate between them well. 

I think it would be fine for us to just do the same thing for line interpolation, but if you wanted to support variable numbers of points for line segments, it would be possible to make that look good. You have to start by making sure that the two lines have the same number of points. The easiest way to do this is by repeating the last point of the shorter list until they're the same length. 

The hard way that produces better results is to insert points between existing ones on the shorter line in a way that attempts to keep the relative distances between points on the two lines similar. Almost certainly not worth it though! but also maybe fun

Reply:
Hmm, thanks for the insight. Will need a bit more clarity to be able to implement what you've suggested, maybe we could have a brief call later? I'll convert this back to a draft in the meantime.